timescaledb/how-to-guides/analytics/about-analytics.md

```diff
@@ -0,0 +1,75 @@
+# About Timescale Analytics
+Timescale Analytics is a PostreSQL extension containing a specialized set of
+functions that allow you to to analyze time-series data. You can use it to
+analyze anything you have stored as time-series data, including IoT devices, IT
+systems, marketing analytics, user behavior, financial metrics, and
+cryptocurrency. Timescale Analytics uses [Promscale][doc-promscale], an
+open-source analytics platform, for Prometheus monitoring data.
+
+Timescale Analytics allows you to perform critical time-series queries quickly,
+analyze time-series data, and extract meaningful information. It aims to
+identify, build, and combine all of the functionality SQL needs to perform
+time-series analysis into a single extension.
+
+## Tools for graphing
+Timescale Analytics brings graphing functions to the database. This allows you
+to choose your graphing front-end based on how well it does graphing, not on how
+well it does data analytics. It also allows you to run queries that stay
+consistent across all front-end tools and consumers of your data. Additionally,
+by doing all the graphing work in the database, you need to send a much smaller
+number of data points over the network.
+
+## Simplifying queries
+SQL queries can get long, especially if you have multiple layers of aggregation
+and function-calls. There are many scenarios where it's possible to write a
+query in native SQL, but the resulting code is relatively complicated to write,
+and to understand. Timescale Analytics can greatly simplify your queries by
+using a two-step calling convention.
+
+For example, a typical Timescale Analytics query to get the time-weighted
+average of a set of values could look like this: ```sql SELECT
```

Comment:
I think we may be conflating a few things here that shouldn't be conflated. This is an explanation around two step aggregation, not really simplification...I'd use some different examples here....

Reply:
Sure. I admit I struggled a little bit here. Do you have some text for me?

Reply:
The current iteration of the blog post has a good before / after query: 
So let’s look first at the normal average versus the time weighted average for this period:
```SQL
SELECT freezer_id, 
	avg(temperature), 
	average(time_weight('Linear', ts, temperature)) as time_weighted_average 
FROM freezer_temps
GROUP BY freezer_id;
```
```output
 freezer_id |  avg  | time_weighted_average 
------------+-------+-----------------------
          1 | 10.35 |     6.802777777777778
```

That gives us a much better picture of what happened. You’ll note that we have to choose an interpolation method, we have two, `Linear` which we chose here and interpolates linearly between the two points and `LOCF` which stands for last observation carried forward, which just takes the previous point as valid until we see a new one. `LOCF` is particularly good for systems which only record when a value changes. 

The main reason we implemented this is that it makes it much easier to do this sort of analysis, here’s what the equivalent query would look like if you wanted to do this in raw SQL:
```SQL
WITH setup AS (
	SELECT lag(temperature) OVER (PARTITION BY freezer_id ORDER BY ts) as prev_temp, 
		extract('epoch' FROM ts) as ts_e, 
		extract('epoch' FROM lag(ts) OVER (PARTITION BY freezer_id ORDER BY ts)) as prev_ts_e, 
		* 
	FROM  freezer_temps), 
nextstep AS (
	SELECT CASE WHEN prev_temp is NULL THEN NULL 
		ELSE (prev_temp + temperature) / 2 * (ts_e - prev_ts_e) END as weighted_sum, 
		* 
	FROM setup)
SELECT freezer_id, 
	avg(temperature),
	sum(weighted_sum) / (max(ts_e) - min(ts_e)) as time_weighted_average 
FROM nextstep
GROUP BY freezer_id;
```
```output
 freezer_id |  avg  | time_weighted_average 
------------+-------+-----------------------
          1 | 10.35 |     6.802777777777778
```

That’s no fun to write, inefficient, and it doesn’t compose well, for instance it doesn’t play nicely with continuous aggregates as it involves window functions and the like. Time weighted average works with continuous aggregates and is a [two-step aggregate](docs link), which means that we can re-aggregate (which would probably be more useful if we had more data here) much the same way we did with the percentile approximation functions above[^twa-reagg-note]. 

I'd probably do a before / after like that to demonstrate the simplification part (or maybe link to it?)